test/jdk/java/net/httpclient/http2/PushPromiseContinuation.java

```diff
@@ -0,0 +1,231 @@
+            }
+
+            for (int i = 0; i < 10; i++) {
+                setPushHeaders("x-push-header-" + i, "data_" + i);
+                setContHeaders("x-cont-header-" + i, "data_" + i);
+            }
+            HttpHeaders pushPromiseHeaders = pushPromiseHeadersBuilder.build();
+            HttpHeaders continuationHeaders = continuationHeadersBuilder.build();
+            testHeaders = testHeadersBuilder.build();
+
+            // Create the Push Promise Frame
+            OutgoingPushPromise pp = new OutgoingPushPromise(streamid, uri, pushPromiseHeaders, content);
+            // Indicates to the client that a continuation should be expected
+            pp.setFlag(0x0);
+
+            // Create the continuation frame
+            List<ByteBuffer> encodedHeaders = conn.encodeHeaders(continuationHeaders);
+            ContinuationFrame cf = new ContinuationFrame(streamid, HeaderFrame.END_HEADERS, encodedHeaders);
+
+            try {
+                // Schedule push promise and continuation for sending
+                conn.outputQ.put(pp);
+                conn.outputQ.put(cf);
+            } catch (IOException ex) {
+                System.err.println("TestServer: pushPromise exception: " + ex);
+            }
+        }
+    }
+
+    static class ServerPushHandler implements Http2Handler {
+
+        private final String mainResponseBody;
+        private final String mainPromiseBody;
+
+        public ServerPushHandler(String mainResponseBody,
+                                 String promise)
+        {
+            Objects.requireNonNull(promise);
+            this.mainResponseBody = mainResponseBody;
+            this.mainPromiseBody = "Main Promise Body";
+        }
+
+        public void handle(Http2TestExchange exchange) throws IOException {
+            System.err.println("Server: handle " + exchange);
+            try (InputStream is = exchange.getRequestBody()) {
+                is.readAllBytes();
+            }
+
+            if (exchange.serverPushAllowed()) {
+                pushPromise(exchange);
+            }
+
+            // response data for the main response
+            try (OutputStream os = exchange.getResponseBody()) {
+                byte[] bytes = mainResponseBody.getBytes(UTF_8);
+                exchange.sendResponseHeaders(200, bytes.length);
+                os.write(bytes);
+            }
+        }
+
+        static final BiPredicate<String,String> ACCEPT_ALL = (x, y) -> true;
+
+
+        private void pushPromise(Http2TestExchange exchange) throws IOException {
+            URI requestURI = exchange.getRequestURI();
+            URI uri = requestURI.resolve("/promise");
+            InputStream is = new ByteArrayInputStream("Test_String".getBytes(UTF_8));
+            Map<String, List<String>> map = new HashMap<>();
+            map.put("x-promise", List.of(mainPromiseBody));
+            HttpHeaders headers = HttpHeaders.of(map, ACCEPT_ALL);
+            exchange.serverPush(uri, headers, is);
```

Comment:
Could create all of the headers here instead of in Http2LPPTestExchangeImpl, might improve readability of test.